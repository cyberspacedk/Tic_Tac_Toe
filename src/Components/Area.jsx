import React from "react";
import Cell from "./Cell";
import styled from "styled-components";

const AreaWrapper = styled.div`
  display: flex;
  flex-wrap: wrap;
  width: 300px;
  button:hover {
    background-color: #ccc;
  }
`;

const Area = ({ cells, onClick }) => (
  <AreaWrapper> 
    {cells.map((cell, idx) => (
      <Cell value={cell} onClick={() => onClick(idx)} key={Date.now() + idx} />
    ))}
  </AreaWrapper>
);

export default Area;
