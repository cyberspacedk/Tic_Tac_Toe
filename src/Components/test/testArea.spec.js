import React from "react";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import Area from "../Area";

describe("container for cells", () => {
  // 1- snapshot
  // 2- check. count of cells
  // 3- check. is cell have prop value
  // 4- check. is cell have prop onClick
  

  const props = {
    cells: Array(9).fill("")
  };

  const myComponent = shallow(<Area {...props} />);

  it("Area snapshot", () => {
    expect(shallowToJson(myComponent)).toMatchSnapshot();
  });

  it("Component should draw 9 <Cell />", () => { 
    expect(myComponent.find("Cell")).toHaveLength(9);
  });

  it("Check. Is cell have prop value", () => {
    expect(myComponent.find('Cell').first().prop('value')).toEqual('')
  });

  it("Check. Is cell have prop onClick", () => {
    expect(myComponent.find('Cell').first().prop('onClick')).toBeTruthy()
  });
});
