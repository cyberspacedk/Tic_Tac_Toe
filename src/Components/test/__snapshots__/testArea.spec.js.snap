// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`container for cells Area snapshot 1`] = `
<styled.div>
  <Cell
    key="0"
    onClick={[Function]}
    value=""
  />
  <Cell
    key="1"
    onClick={[Function]}
    value=""
  />
  <Cell
    key="2"
    onClick={[Function]}
    value=""
  />
  <Cell
    key="3"
    onClick={[Function]}
    value=""
  />
  <Cell
    key="4"
    onClick={[Function]}
    value=""
  />
  <Cell
    key="5"
    onClick={[Function]}
    value=""
  />
  <Cell
    key="6"
    onClick={[Function]}
    value=""
  />
  <Cell
    key="7"
    onClick={[Function]}
    value=""
  />
  <Cell
    key="8"
    onClick={[Function]}
    value=""
  />
</styled.div>
`;
